pyo3-introspection/src/model.rs

```diff
@@ -14,4 +14,27 @@ pub struct Class {
 #[derive(Debug, Eq, PartialEq, Clone, Hash)]
 pub struct Function {
     pub name: String,
+    pub arguments: Vec<Argument>,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Hash)]
+pub struct Argument {
+    pub name: String,
+    pub kind: ParameterKind,
```

Comment:
Not sure if we should use this `kind` enum or a set of different list of arguments (posonlyargs, args, vararg, kwonlyargs, kw_defaults, kwarg) like the [Python AST module](https://docs.python.org/3/library/ast.html).

Reply:
The advantage of having different lists of arguments would be that it guarantees the structure of parameters is ok. Otherwise the parameters might come in a confusing order. Maybe something like:

```rust
positional_args: Vec<String>,
positional_only_args: usize, // within range 0..positional_args.len()
positional_defaults: Vec<String>, // at most has positional_args.len() elements
vararg: Option<String>,
keyword_only_parameters: Vec<(String, Option<String>)>, // keyword, maybe with default
kwarg: Option<String>
```

This would have reasonable alignment with the `FunctionDescription` type internal to PyO3's argument parsing, which might make it easier to generate.

Reply:
Thank you! I have done it in a slightly different way, still using an `Argument` object (useful to store both the default value and the type annotation). Happy to tweak it again if you want.